### PR TITLE
feat(nuxt): add scalar tab to nuxt devtools

### PR DESCRIPTION
**Pull request by arashsheyda**
**Problem**
Currently, there is not scalar tab in Nuxt DevTools

**Solution**
this PR adds support for Nuxt DevTools

### CODE REVIEW

File: packages/nuxt/src/module.ts [hunk 1]
```diff
@@ -71,6 +72,21 @@ export default defineNuxtModule<ModuleOptions>({
       })
     })
 
+    // add scalar tab to DevTools
+    if (_nuxt.options.dev && _options.devtools) {
+      _nuxt.hook('devtools:customTabs', (tabs) => {
+        tabs.push({
+          name: 'scalar',
+          title: 'Scalar',
+          icon: 'https://scalar.com/favicon.svg',
```
Comment (arashsheyda) [inline packages/nuxt/src/module.ts:81]: just realized the logo would disappear in light mode	
Screenshot 2024-04-22 at 1 42 11 PM

**Comment by amritk** [comment]
Hey @arashsheyda thanks for the PR! Didn't realize it was that easy to add devtools support. Whats the best way to get the icon colored correctly? We have the [svg too](https://github.com/scalar/scalar/blob/main/packages/components/src/components/ScalarIcon/icons/Logo.svg). 

Also for the typescript error, if you could just add `_options.pathRouting` to your if statement

```ts
    // add scalar tab to DevTools
    if (_nuxt.options.dev && _options.devtools && _options.pathRouting) {
```

its defaulted anyway

**Comment by arashsheyda** [comment]
@amritk yes devtools api is so good!

for the icon when we use a link it adds it as an image element so there's not much that we can do about that.

however, devtools uses [iconify](https://icon-sets.iconify.design/) so maybe if we could add scalar's svg logo to one of iconify's collection (`logos` sound like a good option), then we can simply say `i-logos-scalar` and that would work!

**Comment by amritk** [comment]
re: typescript weird, ya that works! 

tagging in @cameronrohani for [iconify](https://icon-sets.iconify.design/)

**Comment by cameronrohani** [comment]
Let me know if this works :) 

https://gist.github.com/cameronrohani/0fa020f6dcf957266bff49e7b6b7c05e

**Comment by arashsheyda** [comment]
@amritk I don't know if I did right or not (and sorry I forgot to read the contribution guide 👀)

**Comment by amritk** [comment]
@arashsheyda no problem I forget all the time 😁

**Comment by arashsheyda** [comment]
just saw this [verion of logo](https://private-user-images.githubusercontent.com/6176314/324625960-c7976886-0928-4538-b8d4-61165e3a65ca.svg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM4MjQyNTgsIm5iZiI6MTcxMzgyMzk1OCwicGF0aCI6Ii82MTc2MzE0LzMyNDYyNTk2MC1jNzk3Njg4Ni0wOTI4LTQ1MzgtYjhkNC02MTE2NWUzYTY1Y2Euc3ZnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjJUMjIxMjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2JkNGJiNzYzYTgyYjJmZDE4MjZhODBkZjU0MDQyMGJkOWQ1YWM2MzE3NTkwMGNmODRlNmYzMWUzYmVmNzgyZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.kOb2Ohvn4KmZoLBsI8UlWKIDXUBSMcrGBXdL8OmoTOQ)

how about using the same color?

**Comment by cameronrohani** [comment]
> just saw this [verion of logo](https://private-user-images.githubusercontent.com/6176314/324625960-c7976886-0928-4538-b8d4-61165e3a65ca.svg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM4MjQyNTgsIm5iZiI6MTcxMzgyMzk1OCwicGF0aCI6Ii82MTc2MzE0LzMyNDYyNTk2MC1jNzk3Njg4Ni0wOTI4LTQ1MzgtYjhkNC02MTE2NWUzYTY1Y2Euc3ZnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjJUMjIxMjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2JkNGJiNzYzYTgyYjJmZDE4MjZhODBkZjU0MDQyMGJkOWQ1YWM2MzE3NTkwMGNmODRlNmYzMWUzYmVmNzgyZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.kOb2Ohvn4KmZoLBsI8UlWKIDXUBSMcrGBXdL8OmoTOQ)
> 
> how about using the same color?

added to the gist, let me know if this works? :)  https://gist.github.com/cameronrohani/0fa020f6dcf957266bff49e7b6b7c05e

**Comment by arashsheyda** [comment]
LGTM, what do you think?

Screenshot 2024-04-22 at 4 36 42 PM
Screenshot 2024-04-22 at 4 36 53 PM

File: packages/nuxt/src/module.ts [hunk 2]
```diff
@@ -71,6 +72,21 @@ export default defineNuxtModule<ModuleOptions>({
       })
     })
 
+    // add scalar tab to DevTools
+    if (_nuxt.options.dev && _options.devtools) {
+      _nuxt.hook('devtools:customTabs', (tabs) => {
+        tabs.push({
+          name: 'scalar',
+          title: 'Scalar',
```
Comment (arashsheyda) [inline packages/nuxt/src/module.ts:80]: one other thing that I forgot to mention, do you want to add `scalar` to `server` category? cc @amritk

Comment (amritk) [reply packages/nuxt/src/module.ts:80]: yep that sounds like a good spot for it!

**Comment by amritk** [comment]
Looks great! Thanks for the quick work @arashsheyda

**Review by amritk — APPROVED** [review]

**Review by marclave — APPROVED** [review]
AWESOME WORK @arashsheyda ✨✨

**Pull request merged**